### PR TITLE
feat(sol): initial implementation of explain transaction

### PR DESCRIPTION
**Pull request by joey-bai**
partially implements explain transaction for solana.
does not handle fees or transaction ids correctly.

Ticket: STLX-8791

Note: follows new standardized pattern, where implementation is in account-lib.
See Dot PR for reference: https://github.com/BitGo/BitGoJS/pull/1723/files

### CODE REVIEW

File: modules/account-lib/src/coin/sol/transaction.ts [hunk 1]
```diff
@@ -188,4 +189,75 @@ export class Transaction extends BaseTransaction {
...
+      // TODO (STLX-8791): Need to figure out what id to use for unsigned transactions.
+      id: this.id || 'undefined',
+      type: TransactionType[this.type].toString(),
+      changeOutputs: [],
+      changeAmount: '0',
+      outputAmount: outputAmount.toFixed(0),
+      outputs: outputs,
+      fee: {
+        // TODO(STLX-8791): need to figure out how to get fee information.
+        fee: '0',
```
Comment (alebusse) [inline modules/account-lib/src/coin/sol/transaction.ts:254]: With just a rawTx there is no way to figure out the fee, we only can extract the number of signatures that is needed to calculate the fee.

**Review by alebusse — APPROVED** [review]

File: modules/account-lib/src/coin/baseCoin/iface.ts [hunk 2]
```diff
@@ -87,3 +87,36 @@ export interface Entry extends BaseAddress {
 export interface BaseFee {
   fee: string;
 }
+
+export interface TransactionRecipient {
```
[the remark this answers is not in the document]
Comment (alebusse) [reply modules/account-lib/src/coin/baseCoin/iface.ts:91]: I think @argjv changed the explain transaction to accountLib as new standard so this is needed for that.

[the remark this answers is not in the document]
Comment (joey-bai) [reply modules/account-lib/src/coin/baseCoin/iface.ts:91]: +1, moving forward, explainTransaction should be implemented in account-lib.
we are doing the same for Dot: https://github.com/BitGo/BitGoJS/pull/1723/files#diff-52364a505ce7631f55f9cfdbc646e903efef4b998d80ab1b34d835206b0da730

File: modules/account-lib/src/coin/sol/transaction.ts (hunk 1 above)
Comment (joey-bai) [reply modules/account-lib/src/coin/sol/transaction.ts:254]: that's true... will ask around to see how we want to handle this for the deserialization flow. For Dot, i see that they have clients set the `fee`, but not sure how they're getting it back on deserialization.

**Pull request merged**